app/models/github_callback.rb

```diff
@@ -27,22 +34,33 @@ def repo
   end
 
   def files_modified
-    event['commits'].map { |commit| commit['added'] + commit['modified'] }.flatten.uniq
+    event['commits'].map { |commit| commit['added'] + commit['modified'] }.flatten.map { |filename| filename.gsub('_cheat.md', '.md').gsub('_teacher.md', '.md') }.uniq
   end
 
   def files_removed
-    event['commits'].map { |commit| commit['removed'] }.flatten.uniq
+    event['commits'].map { |commit| commit['removed'] }.flatten.map { |filename| filename.gsub('_cheat.md', '.md').gsub('_teacher.md', '.md') }.uniq
   end
 
-  def files_touched
-    files_modified + files_removed
+  def layout_files_modified
+    files_modified.select { |path| path.include?('yaml') }
   end
 
-  def directories_updated
-    unique_directories(files_touched)
+  def update_modified_lessons
+    files_modified.each do |file|
+      lessons = Lesson.where(github_path: "https://github.com/#{ENV['GITHUB_CURRICULUM_ORGANIZATION']}/#{repo}/blob/master/#{file}")
+      lessons.each do |lesson|
+        lesson.update_from_github
+        lesson.save
+      end
+    end
   end
 
-  def unique_directories(paths)
-    paths.select { |path| path.include?('/') }.map {|path| path.split('/').first}.uniq
+  def update_removed_lessons
+    files_removed.each do |file|
+      lessons = Lesson.where(github_path: "https://github.com/#{ENV['GITHUB_CURRICULUM_ORGANIZATION']}/#{repo}/blob/master/#{file}")
+      lessons.each do |lesson|
+        lesson.update_columns(public: false)
```

Comment:
It seems weird to use update_columns - why are we not just deleting? If this is the right way to go, `update_all` is probably the better method to use, since you can apply it to the entire query.

Reply:
Hmm this is leftover from the way we were doing things before. I'll check with Franz but now that everything is sourced from Github I imagine it would be fine to just delete the lesson.